core/src/replay_stage.rs

```diff
@@ -3779,7 +3779,7 @@ pub(crate) mod tests {
     fn test_dead_fork_entry_deserialize_failure() {
         // Insert entry that causes deserialization failure
         let res = check_dead_fork(|_, bank| {
-            let gibberish = [0xa5u8; SIZE_OF_DATA_SHRED_PAYLOAD];
+            let gibberish = [0xa5u8; /*legacy data-shred capacity:*/1051];
```

Comment:
> This would make it easier to see all locations where the value is used.

That is indeed useful and necessary for actual uses. But I am not sure if it matters for "tests".

The "capacity" thing is no longer constant, and depends on the shred variant (legacy vs merkle) and proof size in case of merkle variants. So exposing that requires exposing further implementation details that I am not in favor of.

Ideally we should not expose implementation details or add backdoors to rust access rules just for the sake of tests; and instead just test public interfaces outside of the crate which gives larger test coverage because of running more logic.

Here for example, it should have used `shredder` api instead of making shreds itself; and the "capacity" should only be exposed to shredder module.

Reply:
Giving this a second thought, in https://github.com/solana-labs/solana/pull/25631 I added:
```rust
// For legacy tests and benchmarks.
const_assert_eq!(LEGACY_SHRED_DATA_CAPACITY, 1051);
pub const LEGACY_SHRED_DATA_CAPACITY: usize = legacy::ShredData::CAPACITY;
```
not to scatter `1051` through the code;
but going forward I still think the better path is to use public interfaces instead of leaking implementation details or adding backdoors.